protocol/cosi.go

```diff
@@ -213,37 +209,29 @@ func (pc *ProtocolCosi) handleCommitment(in *Commitment) error {
 		return pc.commitmentHook(pc.tempCommitment)
 	}
 
-	// or make continue the cosi protocol
-	commits := make([]*cosi.Commitment, len(pc.tempCommitment))
-	secretVar := pc.Suite().Point().Null()
-	for i := range pc.tempCommitment {
-		secretVar.Add(secretVar, pc.tempCommitment[i].Commitment.Commitment)
-		commits[i] = pc.tempCommitment[i].Commitment
-	}
-
 	// go to Commit()
-	out := pc.Cosi.Commit(nil, commits)
-	secretVar.Add(secretVar, pc.Cosi.GetCommitment())
+	out := pc.Cosi.Commit(nil, pc.tempCommitment)
+
 	// if we are the root, we need to start the Challenge
 	if pc.IsRoot() {
 		return pc.StartChallenge()
 	}
 
 	// otherwise send it to parent
 	outMsg := &Commitment{
-		Commitment: out,
+		Comm: out,
 	}
 	return pc.SendTo(pc.Parent(), outMsg)
 }
 
 // StartChallenge start the challenge phase. Typically called by the Root ;)
-func (pc *ProtocolCosi) StartChallenge() error {
+func (pc *CoSi) StartChallenge() error {
```

Comment:
In fact I can't see how you would merge the two without changing the underlying `dedis/crypto/cosi` because, in `StartChallenge` it's `CreateChallenge`, and in `handleChallenge` it uses `cosi.Challenge`. Both functions don't do the same, while in the `Announcement` phase the behavior is the same.